File: dataset/codenet/Java/Code/s393629232.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int maxv = Integer.MIN_VALUE;
        int minv = sc.nextInt();
        for (int i = 1; i < n; i++) {
            int ri = sc.nextInt();
            maxv = Math.max(maxv, ri - minv);
            minv = Math.min(minv, ri);
        }
        System.out.println(maxv);
        sc.close();
    }
}

File: output/deepseek_chat/codenet/algo_based_translation/java/python/s393629232.py

import sys

def main():
    n = int(input())
    maxv = -sys.maxsize - 1
    minv = int(input())
    for i in range(1, n):
        ri = int(input())
        maxv = max(maxv, ri - minv)
        minv = min(minv, ri)
    print(maxv)

if __name__ == "__main__":
    main()

